### ISSUE TITLE
Drag and Drop functionality

### ISSUE THREAD

**Issue by zim0369** [flameshot-org/flameshot#2899]
### Feature Description

This feature would be different to what flameshot is intended for but it woulde be nice to extend flameshot be able to drag and drop images from the terminal. More specifically, something like `flameshot recent` would open a small floating window with a preview of the recently captured screenshots with an ability to drag and drop them.

**Comment by mmahmoudian** [comment]
@zim0369  I agree that this is out of the context of Flameshot. Perhaps you might be interested in this thread: https://github.com/flameshot-org/flameshot/issues/556#issuecomment-952244072

How about closing this issue in favor of #556 ?

**Comment by zim0369** [comment]
@mmahmoudian although this feature is out of context, IG it would be a lot more convenient to have it.

It's finally up to you.

**Comment by mmahmoudian** [comment]
@zim0369 the fact is this can be easily scripted even in bash using various typical tool like fzf. 

But considering the discussion in that thread I mentioned above, I believe there are some merits in this. Did you managed to read that thread?

**Comment by zim0369** [comment]
@mmahmoudian yes, i did.

This could be scripted but it wouldn't be exaclty the same. But, since it could be scripted I think it makes this issue redundant.

**Comment by mmahmoudian** [comment]
I'm interested to know the opinion of others, esp. @veracioux and @borgmanJeremy on this matter.

**Comment by zim0369** [comment]
Okay @mmahmoudian

I went through that issue once again and I can confirm that it isn't close to what I'm describing.

**A verbose description:** 
Entering `flameshot recent` brings up a window that stretches from left edge of  the screen to the right edge. That window has about 1/4th height of that of the screen and starts at the bottom or the top. 

This preview window has recent images arranged horizontally. The borders are semi-transparent, matching flameshot's UI. Hovering over an image brings up a copy button over the upper right corner of the image. Clicking on that button copies the full path of the image to the clipboard. After copying the path the user could click on a close button to close the window. 

User can hover over an image and drag it. Dragging an image makes the floating window disappear. The user can then drop the image wherever he wants to.

We could go even further and add select buttons that appear on hovering over an image. They allow users to select multiple images for drag and drop.

**Comment by veracioux** [comment]
@zim0369 @mmahmoudian I think this can and should be a standalone program (let's call it `recents`). Except for the part where `recents` has to know all the images that were created by flameshot. But this can be solved by creating a plugin that hooks into an event just before flameshot exports the capture (via save, copy or any other final action) and then calls `recents` to let it know that a screenshot was taken by flameshot.

I would be against integrating any of these features into vanilla flameshot though.

**Comment by mmahmoudian** [comment]
@veracioux 

> But this can be solved by creating a plugin that hooks into an event just before flameshot exports the capture

You are right, the plugin is a perfect place for this.
